data_structures/queue_implementation.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @author [saqlain] (https://github.com/Mysterious786)
+ * @file
+ *
+ * Implementation of a Queue using Single Stacks.
+ *
+ */
+
+#include <iostream>  /// for IO operations
+#include <stack>     /// for stack operations
+#include <cstdlib>   /// for datatype conversion
+namespace
+{
+    /**
+     * Queue data structure.Store elements in FIFO
+     * (first-in-first-out) manner.
+     *
+     */
+
+    class Queue
+    {
+        std::stack<int> s;
+
+    public:
+        /**
+         * Pushes data to the back of queue
+         */
+        void enqueue(int data)
+        {
+            /**
+             *  Push item into the First Stack
+             */
+            s.push(data);
+        }
+
+        /**
+         * Remove an item from the queue
+         */
+        int dequeue()
+        {
+            /**
+             * Determines whether the stack is empty
+             */
+            if (s.empty())
+            {
+                std::cout << "Underflow!!";
+                exit(0);
+            }
+
+            /**
+             *Removes item from the Stack
+             */
+            int top = s.top();
+            s.pop();
+
+            
+            /**
+             * Determines whether the stack becomes empty,and returns the popped item
+             */
+            if (s.empty())
+            {
+                return top;
+            }
+
+            /**
+             * Recursion call
+             */
+            int item = dequeue();
+
+            /**
+             * push popped item back into the stack
+             */
+            s.push(top);
+
+            
+            /**
+             * return the result of dequeue() call
+             */
+            return item;
+        }
+    };
+}
```

Comment:
Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or on our [**Discord server**](https://the-algorithms.com/discord). 😃